app/assets/javascripts/preferences/components/PreferencesPane.tsx

```diff
@@ -14,14 +14,16 @@ export const PreferencesSegment: FunctionComponent = ({ children }) => (
 
 export const PreferencesGroup: FunctionComponent = ({ children }) => (
   <div className="bg-default border-1 border-solid rounded border-gray-300 px-6 py-6 flex flex-col gap-2">
-    {!Array.isArray(children)
+    {Array.isArray(children)
       ? children
-      : children.map((c, i, arr) => (
-          <>
-            {c}
-            <HorizontalLine index={i} length={arr.length} />
-          </>
-        ))}
+          .filter((c) => c != null && c !== '')
```

Comment:
Nitpick, but let's explicitly use strict equality comparisons. And maybe more descriptive variable names.
```suggestion
          .filter((child) => child !== null && child !== '')
```

Reply:
Changed c to child, but the comparison `!=null` compares both whether the variable is null or undefined, so it's fine like that. That's why the TS compiler doesn't complain.
```
null != null => false;
undefined != null => false;
```

Reply:
Hmm, in that case I would prefer to compare against both `null` and `undefined` explicitly or have some utility function which does that in case there are many such checks. 
Having written it explicitly would actually improve the readability, also the code will be more consistent, since here we are checking the value against `null`, but in `TwoFactorAuth` - against `undefined`.

Reply:
There is a utility function in snjs already, but it's a feature of the language, it's part of the syntax, we don't need a utility function. I've changed it to undefined, for consistency.